service/collector/trusted_advisor.go

```diff
@@ -11,6 +13,7 @@ import (
 	"golang.org/x/sync/errgroup"
 
 	"github.com/giantswarm/aws-collector/client/aws"
+	"github.com/giantswarm/aws-collector/service/internal/cache"
```

Comment:
Speaking from a software design point of view the caching should maybe be transparent. Introducing caching to existing functionality would at best mean no or only little change in the existing implementation. So in `service/collector/trusted_advisor.go` not much should change, though right now here are most of the changes. That means business logic and caching is mixed which means it makes everything more complicated and harder for anyone working on the code in the future. A better design could mean to implement the caching separately and inject it in the existing implementation transparently. We can talk about this if you are interested.

Reply:
Yeah we can definitely talk about refactoring the cache here. I also don't like it that much.